docs/rfcs/xxxx-backwards-compatibility.md

```diff
@@ -0,0 +1,111 @@
+# Grafana Agent backwards compatibility 
+
+* Date: 2023-05-25
+* Author: Robert Fratto (@rfratto)
+* PR: [grafana/agent#3981](https://github.com/grafana/agent/pull/3981)
+* Status: Draft
+
+Grafana Agent has been following [semantic versioning](https://semver.org/) since its inception.
+After three years of development and 33 minor releases, the project is on trajectory to have a 1.0 release. 
+
+In the context of semantic versioning, a 1.0 release indicates that future minor releases have backwards compatibility with older minor releases in the same major release; version 1.1 is backwards compatible with version 1.0. Having major and minor releases signals to users when an upgrade may take more time (major releases) and when they can upgrade with more confidence (minor releases).
+
+However, Grafana Agent is a large project, with a large surface area for what may be considered part of the backwards compatibility guarantees. This proposal formally establishes what parts of Grafana Agent will be protected by backwards compatibility. 
+
+## Goals 
+
+- Set expectations for what is covered by backwards compatibility. 
+- Set expectations for when upgrades to new major versions will be forced.
+
+## Proposal 
+
+Backwards compatibility means that a user can upgrade their version of Grafana Agent without needing to make any changes with the way they interact with Grafana Agent, provided that interaction is within scope of being covered by backwards compatibility.
+
+### Scope of backwards compatibility   
+
+The following will be protected by backwards compatibility between minor releases: 
+
+- **User configuration**, including configuration file syntax and the command-line interface. 
```

Comment:
Based on discussions in other threads, I think we should also be clear that functional meaning / semantics of the configuration should also be covered - for CLI, API and config files. 

Imagine a change where `loki.write` starts to just print to stdout, like `loki.echo` - that would break users' telemetry pipelines, even though it may not break the config files.